src/api/firebase.js

```diff
@@ -72,10 +73,66 @@ export async function addItem(listId, { itemName, daysUntilNextPurchase }) {
 	});
 }
 
-export async function updateItem(listId, itemId, currentItemUpdates) {
+export async function updateItem(
+	listId,
+	itemId,
+	dateLastPurchased,
+	dateNextPurchased,
+	dateCreated,
+	currentItemUpdates,
+) {
 	const currentItemRef = doc(db, listId, itemId);
+	// use updated totalPurchases
+	const { totalPurchases } = currentItemUpdates;
+
+	let daysSinceLastTransaction;
+	let previousEstimate;
+	// When the item was purchased first time
+	if (totalPurchases === 1) {
```

Comment:
**No need to change**

Relying on comments sometimes is not the best solution (what you have here works btw) because it means that whenever the code changes, two things could happen:
1. Ideally, you update the code, then you have to update the comment. That's two things you have to do.
2. **Not ideal** You update the code but forget to update the comment. 

That said, you can eliminate comments like this one by using variables with meaningful names, for instance, I think that this if/else statement could be written like:

```js
// pseudo code
const isFirstTimePurchase = totalPurchases === 1;

if (isFirstTimePurchase) {
  /** implementation */
} else {
  /** implementation */
}
```

There, you don't need the comment above the `if` statement because `isFirstTimePurchase` is already telling you what condition you're checking for, and therefore, you don't need the comment above the `else` statement because you can infer that `isFirstTimePurchase === false,` then it means that this is the second+ time the purchase happens.

You don't need to change this to merge this PR, keep it in mind for the future though, and ask yourself "Is there a way to omit this comment I just wrote?" sometimes the answer is YES. On the other hand, sometimes you have to resort to comments, and that's OK, too. But the comments usually should be used (or at least, this is what I do) to describe WHY something is happening instead of WHAT the code is doing.

For this if/else statement I would imagine that you could write something like

```js
/**
 We need to use a new date only when the user buys the item for the first time
 because ....

 Otherwise, we need to do ... because of [business reasons]
*/
if (/** condition */) {
  /** implementation */
} else {
  /** implementation */
}
```

Reply:
Thanks for the tips on the comment and how to use variables to explain code.  I just kept the comment here for the reviewers since it was confusing to me. I will delete them too.